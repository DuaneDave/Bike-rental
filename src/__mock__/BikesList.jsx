import React from "react"

const BikesList = () => {

    const isLoading = false;
    const isSuccess = true;
    const isError = false;


    const bikes = [
       { id: 1,
        name: "Bike",
        brand: "Brand",
        bike_type: "Type",
        daily_rate: 100,
        description: "Description",
        color: ["red", "blue", "green"],
        images: {
          red: "https://images.unsplash.com/photo-1610000000000-000000000000?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60",
          blue: "https://images.unsplash.com/photo-1610000000000-000000000000?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60",
          green: "https://images.unsplash.com/photo-1610000000000-000000000000?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60",
        },
    }
    
      ]
    

  return (

    <>
      <div className="swiper-button image-swiper-button-next">
      </div>
      <div className="swiper-button image-swiper-button-prev">
      </div>
      {isLoading && <Spiner />}
      {isSuccess &&
        bikes.map((bike) => (
          <p  key={bike.id}>
            <img
              onClick={() => navigate(`/Bikes/${bike.id}`)}
              src={bike.images[Object.keys(bike.images)[0]]}
            ></img>
            <h3>
              {bike.name} {bike.brand}
            </h3>
            <p>
              {" "}
              {bike.description.substring(1, 100)}
              <span
                style={{ cursor: "pointer" }}
                onClick={() => navigate(`/Bikes/${bike.id}`)}
              >
                ...
              </span>
            </p>
          </p>
        ))}
      {isError && error.toString()}
      </>
  );
};

export default BikesList;
