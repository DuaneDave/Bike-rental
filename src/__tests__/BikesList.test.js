import renderer from 'react-test-renderer';
import React from 'react';
import BikesList from '../__mock__/BikesList';

describe('Bikes', () => {
  test('if bikes mocked automatically', () => {
    const component = renderer.create(
      <BikesList />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
