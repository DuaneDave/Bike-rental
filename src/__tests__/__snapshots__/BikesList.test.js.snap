// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bikes if bikes mocked automatically 1`] = `
[
  <div
    className="swiper-button image-swiper-button-next"
  />,
  <div
    className="swiper-button image-swiper-button-prev"
  />,
  <p>
    <img
      onClick={[Function]}
      src="https://images.unsplash.com/photo-1610000000000-000000000000?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60"
    />
    <h3>
      Bike
       
      Brand
    </h3>
    <p>
       
      escription
      <span
        onClick={[Function]}
        style={
          {
            "cursor": "pointer",
          }
        }
      >
        ...
      </span>
    </p>
  </p>,
]
`;
